src/engine/mod.rs

```diff
@@ -564,6 +564,36 @@ where
         Ok(None)
     }
 
+    fn execute_write(&mut self) -> Result<Option<Bug>, EngineError> {
+        if !matches!(self.regs[Register::A0 as usize], Value::Concrete(1)) {
+            return not_supported("can not handle other fd than stdout in write syscall");
+        }
+
+        let buffer = if let Value::Concrete(b) = self.regs[Register::A1 as usize] {
+            b
+        } else {
+            return not_supported(
+                "can not handle symbolic or uninitialized buffer address in write syscall",
+            );
+        };
+
+        let size = if let Value::Concrete(s) = self.regs[Register::A2 as usize] {
+            s
+        } else {
+            return not_supported("can not handle symbolic or uinitialized size in write syscall");
+        };
+
+        trace!("write: fd={} buffer={:#x} size={}", 1, buffer, size,);
+
+        if !self.is_in_vaddr_range(buffer) || !self.is_in_vaddr_range(buffer + size) {
+            return not_supported("write syscall failed to access buffer");
+        }
+
```

Comment:
I think we should probably try to actually access the buffer, just to see if there are "Uninitialized" values in there. Usually that's defined as undefined behaviour and therefore I think we should interpret it is a bug. (https://github.com/cksystemsgroup/monster/blob/main/src/engine/bug.rs#L14)

Reply:
Good point. I'll add a loop that scans for uninitialized words in the given buffer.

One quirk I see here is that there will potentially be a strange interaction with the way `read` handles partially initialized words. IIUC, then `read` will mark the entire word as uninitialized if at least one byte remains uninitialized. So the following C* snippet will always fail I think:

```
uint64_t* x;
x = malloc(8);
read(0, x, 7);
write(1, x, 7);
```

Do you have thoughts on that? I am unclear what the semantics of partially initialized words should be here.

Reply:
I decided to define partially initialised words as uninitialised, because it makes everything simpler as long as we do not support other datatypes. But Christoph wants us to extend the support to full blown RISC-V, where this problem has to be resolved by then anyway.

Reply:
Done. I have implemented the scan for (partially) uninitialized words, which will be reported as `Bug::AccessToUnitializedMemory` now.

Reply:
perfect